### PR TITLE
WOR-231: Display validations/warnings from concepts

### PR DESCRIPTION
**Pull request by jomadoye**

### CODE REVIEW

**Comment by mogoodrich** [comment]
@jomadoye do you have an example of how these filters would be used?

**Comment by jomadoye** [comment]
Yes @mogoodrich , I'm already using them for the BloodPressure form 
This is the PR.
https://github.com/PIH/openmrs-pwa-workflow/pull/50/files

ideally 
```
const systolicBloodPressureNormalRange = formValidations.getNormalRangeFromConcept(CONCEPTS.SystolicBloodPressure);
const systolicBloodPressureAbNormalRange = formValidations.getAbNormalRangeFromConcept(CONCEPTS.SystolicBloodPressure);
```

and 
```
validate={props.validate || systolicBloodPressureNormalRange}
 warn={props.warn || systolicBloodPressureAbNormalRange}
```

**Comment by mogoodrich** [comment]
@jomadoye yeah, I found the use case in the Blood Pressure form after I made this comment... :)  I have some updated comments there.

File: src/features/form/validations.js [hunk 1]
```diff
@@ -20,6 +20,42 @@ const dateToInt = dateStr => new Date(dateStr).getTime();
...
+  const {
+    hiAbsolute,	
+    lowAbsolute,
+  } = concept;
+  let hiAbsoluteRange, lowAbsoluteRange;
+
+  if (hiAbsolute || lowAbsolute) {
+    hiAbsoluteRange = hiAbsolute ? maxValue(hiAbsolute) : undefined;
+    lowAbsoluteRange = hiAbsolute ? minValue(lowAbsolute) : undefined;
+    return [hiAbsoluteRange, lowAbsoluteRange].filter(Boolean);
```
Comment (mogoodrich) [inline src/features/form/validations.js]: @jomadoye what does filter(Boolean) do?  Filtered out undefined?

File: src/features/form/validations.js [hunk 2]
```diff
@@ -20,6 +20,42 @@ const dateToInt = dateStr => new Date(dateStr).getTime();
...
+  } else {
+    return [];
+  }
+};
+
+const getAbNormalRangeFromConcept = concept => {
```
Comment (mogoodrich) [inline src/features/form/validations.js]: @jomadoye minorpoint, abnormal is one word, so the "N" should not be capitalized.

File: src/features/form/validations.js (hunk 1 above)
Comment (jomadoye) [reply src/features/form/validations.js]: yes it filters out any falsy value

File: src/components/form/Obs.jsx [hunk 3]
```diff
@@ -152,11 +160,11 @@ const mapStateToProps = (state, props) => {
   const fullPath = (props.obsGroupContext ? props.obsGroupContext.path + '^' : '') + props.path;
 
   const name = formUtil.obsFieldName(fullPath, concepts);
-
+  const concept = selectors.getConcept(state, getUuid(props.concept));
   return {
     name: name,
     value: props.formContext ? props.formContext.selector(state, name) : null,
-    concept: selectors.getConcept(state, getUuid(props.concept)),
+    concept: { ...concept, ...props.concept },
```
Comment (mogoodrich) [inline src/components/form/Obs.jsx]: @jomadoye cool, it look me a second to understand what is happening here, but I understand now... a concise way to allow users to override via props, nice!

**Pull request merged**